tests/framework/web/UrlManagerCreateUrlTest.php

```diff
@@ -758,4 +759,34 @@ public function testUrlCreateCacheWithParameterMismatch()
         $this->assertEquals(2, $rules[0]->createCounter);
         $this->assertEquals(1, $rules[1]->createCounter);
     }
+
+    /**
+     * @see https://github.com/yiisoft/yii2/issues/14406
+     */
+    public function testCreatingRulesWithDifferentRuleConfigAndEnabledCache()
+    {
+        $this->mockWebApplication([
+            'components' => [
+                'cache' => ArrayCache::className()
+            ]
+        ]);
+        $urlManager = $this->getUrlManager([
+            'cache' => 'cache',
+            'rules' => [
+                '/' => 'site/index'
+            ]
+        ]);
+
+        $cachedUrlManager = $this->getUrlManager([
+            'cache' => 'cache',
+            'ruleConfig' => [
+                'class' => CachedUrlRule::className()
+            ],
+            'rules' => [
+                '/' => 'site/index'
+            ]
+        ]);
+
+        $this->assertInstanceOf(CachedUrlRule::className(), $cachedUrlManager->rules[0]);
```

Comment:
it would be more logical to verify that this $rule is not similar to that of $urlManager